torchao/prototype/quantization/safetensors_JSONEncoders.py

```diff
@@ -0,0 +1,175 @@
+    QuantizeTensorToFloat8Kwargs,
+)
+
+ALLOWED_QUANT_DTYPES = {
+    "torch.float8_e4m3fn": torch.float8_e4m3fn,
+    # add to me
+}
+ALLOWED_GRANUALARITY = {"PerRow()": PerRow()}
+
+ALLOWED_AO_MODULES = {
+    "torchao.quantization",
+    "torchao.sparsity.sparse_api",
+    "torchao.prototype.quantization",
+    "torchao.prototype.mx_formats",
+    "torchao.dtypes",
+    "torchao.prototype.awq",
+    "torchao.quantization.quantize_.common",
+    "torchao.quantization.quantize_.workflows",
+}
+
+
+class Float8TensorAttributeJSONEncoder(json.JSONEncoder):
+    def default(self, o):
+        if isinstance(o, Float8Tensor):
+            tensor_attr_dict = {}
+
+            for tensor_attribute_name in o.optional_tensor_attribute_names:
+                attribute = getattr(o, tensor_attribute_name)
+                serialized_attribute = self.encode_value(attribute)
+                if serialized_attribute:
+                    tensor_attr_dict[tensor_attribute_name] = serialized_attribute
+
+            return {
+                "_type": o.__class__.__name__,
+                "_data": tensor_attr_dict
+            }
+
+        if hasattr(o, "_fields") and hasattr(
+            o, "_asdict"
+        ):  # Check for NamedTuple characteristics
+            asdict_data = o._asdict()
+            # Process each field to handle nested objects
+            processed_data = {k: self.encode_value(v) for k, v in asdict_data.items()}
+
+            return {
+                "_type": o.__class__.__name__,
+                "_data": processed_data,
+            }
+
+        if dataclasses.is_dataclass(o) and not isinstance(o, type):
+            data_dict = {}
+            # Process each field to handle nested objects
+            for f in dataclasses.fields(o):
+                if f.name != "version":
+                    data_dict[f.name] = self.encode_value(getattr(o, f.name))
+
+            return {
+                # Only store the class name for dataclasses too
+                "_type": o.__class__.__name__,
+                "_data": data_dict,
+            }
+
+        if isinstance(o, torch.dtype):
+            return {"_type": "torch.dtype", "_data": str(o).split(".")[-1]}
+
+        if isinstance(o, enum.Enum):
+            # Store the full path for enums to ensure uniqueness
+            return {"_type": f"{o.__class__.__name__}", "_data": o.name}
+
+        if isinstance(o, list):
+            return [self.encode_value(item) for item in o]
+
+        return super().default(o)
+
+    def encode_value(self, value):
+        """Helper method to recursively encode a value"""
+        # Try to use default for custom type
+        try:
+            # This will handle all our special cases and raise TypeError
+            # if it can't handle the type
+            result = self.default(value)
+            return result
+        except TypeError:
+            pass
+
+        return value
+
+
+def config_from_dict(data: Dict[str, Any]) -> Float8Tensor:
```

Comment:
this should not be `config_from_dict` I think, seems like `object_from_dict` is more appropriate